documentation/spike/alert_emaails_spike.md

```diff
@@ -0,0 +1,84 @@
+# Alert Email
+
+## Background
+We currently send a huge number job alerts and we want to review this feature in anticipation of an increase once we expand into FE.
+
+We have no visibility on the number of job alert emails that are actually opened.
+
+## Desired outcomes
+Have a stable alert emails jobs memory and comptutationally wise even after the increased number of live vacancies.
+
+## Proposed suggestions
+After reviewing different alternatives to the issue at hand, the following set of suggestions seems to best fit TV.
+They could be applied sequentially without having to commit to a massive task at once.
+
+### Add alert email jobs monitoring
+This is to be able to observe the health of the service. We should monitor:
+- time taken to complete whole job
+- memory used to complete whole job
+- list the N slowest subscriptions to process
+- setup alerting when certain metric's threshold are reached
+
+### Pre-computed subscription.vacancies_matching
+Split subscription.vacancies_matching form email sending to have a predictable behaviour for the email sending even at peak times.
+
+The output of the pre-computation job should be optimized for retrieval so that the subsequent email sending job is as fast as possible.
```

Comment:
I'm not sure what you mean by the 'pre-computation job' ? The alert mailer job is already optimized at the end of last year